### DISCUSSION TITLE
[FEATURE REQUEST] support wildcards in _in requisites

### DISCUSSION THREAD

**Discussion by OrangeDog** [saltstack/salt#67595] [Feature Requests]
> Also, this feature is not consistent (as well as many other things in Salt, unfortunately). It doesn't work with `*_in`-style requisites.

_Originally posted by @max-arnold in https://github.com/saltstack/salt/issues/52308#issuecomment-477129695_

**Describe the solution you'd like**
`require_in` et al. should support the same wildcard syntax as `require` et al.

**Describe alternatives you've considered**
There's no way I know of to get a list of states in the current highstate in order to do this in jinja or anything. There's unlikely to ever be a way, because jinja runs before the highstate is decided. You could possibly do something parsing the result of `state.highstate mock=true` but technically that's not the same run.

The manual alternative is to search for matching states and editing them to add the inverse requisite, which won't work if the intended target isn't always in the highstate (one of the reasons for using `_in` in the first place).

**Comment by frebib** [comment]
I would consider this a bug, mostly because it's inconsistent with the rest of the requisite resolution, and if it was implemented in any sane way, this would _just work_
